### DISCUSSION TITLE
Better blocking performance?

### DISCUSSION THREAD

**Discussion by blipk** [kkapsner/CanvasBlocker#638] [Ideas]
Hi, sometimes there is performance issues on certain pages and Firefox recommends to stop the extension process.

I believe maybe instead of blocking each attempt, you could just rewrite the js to redirect to a null op? Maybe even cache the modifications for visited sites?

**Comment by blipk** [comment]
I guess looking at the settings "Block everything" might do something similar.

**Reply by kkapsner** [reply]
Yes and no. Blocking will block the API access and therefore might break the site.

You can have a look at the rng setting if you want same form of caching. But it does not impact performance much. Caching all JS calls would lead to a massive memory usage and might not even help performance.

The cases where I got that message from Firefox went away when I disabled notifications form DOMRect:
![grafik](https://user-images.githubusercontent.com/1105157/225116344-14e6fcc5-432f-433f-898c-b01a74f96266.png)
